File: Data Structures & Algorithms/max-water-container/submission-7.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0 ; 
        let  r = heights.length - 1; 
        let res = 0 ; 
        while(l <= r){
            let area = Math.min(heights[l],heights[r]) * (r - l);
            if(heights[l] <= heights[r])l +=1;
            else{
                r-=1
            }
            res = Math.max(area,res)
        }
        return res 
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-7.go
func lengthOfLongestSubstring(s string) int {
	set := make(map[string]bool)
	var l = 0
	var res = 0
	for r := 0 ; r < len(s); r++{
		for set[string(s[r])] {
            delete(set, string(s[l]))
            l++
        }
		set[string(s[r])] = true

		res = max(res, r-l+1)
	}
	
	return res
}
